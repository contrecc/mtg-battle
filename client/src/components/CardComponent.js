import React from 'react';
import { Card } from 'reactstrap';
import Image from 'react-graceful-image';

export default props => {
  const { winningCard, card, id, pickedCard } = props;
  let bgClass;

  if (winningCard) {
    if (winningCard === pickedCard && winningCard === id) {
      bgClass = 'bg-success';
    }
    if (winningCard === pickedCard && winningCard !== id) {
      bgClass = 'bg-danger';
    }
    if (winningCard !== pickedCard && winningCard !== id) {
      bgClass = 'bg-danger';
    }
    if (winningCard !== pickedCard && winningCard === id) {
      bgClass = 'bg-success';
    }
  }

  if (!winningCard) {
    bgClass = 'bg-secondary';
  }

  return (
    <Card className={bgClass} style={{ padding: '20px', marginBottom: '30px'}}>
      <Image
        src={card.image_uris.current}
        width="100%"
        alt="card image"
        retry={{ count: 10, delay: 2 }}
        placeholderColor="#6c757d"
        
      />
    </Card>
  );
};

