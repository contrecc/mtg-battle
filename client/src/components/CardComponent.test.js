import CardComponent from './CardComponent';
import Image from 'react-graceful-image';

describe('CardComponent', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <CardComponent
        winningCard="card1"
        card={{ image_uris: { current: 'bluewizard.jpg' } }}
        id="card1"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a div', () => {
    const wrapper = shallow(
      <CardComponent
        winningCard="card1"
        card={{ image_uris: { current: 'bluewizard.jpg' } }}
        id="card1"
      />
    );
    const div = wrapper.find('div').first();
    expect(div.length).toBeGreaterThan(0);
  });

  it('renders a child component of Image', () => {
    const wrapper = shallow(
      <CardComponent
        winningCard="card1"
        card={{ image_uris: { current: 'bluewizard.jpg' } }}
        id="card1"
      />
    );
    const imageComponent = wrapper.find(Image);
    expect(imageComponent.length).toEqual(1);
  });

  it('properly assigns `bg-success` to `bgClass` if card wins', () => {
    const wrapper = shallow(
      <CardComponent
        winningCard="card1"
        card={{ image_uris: { current: 'bluewizard.jpg' } }}
        id="card1"
      />
    );
    const div = wrapper.find('div').first();
    expect(div.find('bg-success')).toBeTruthy();
  });

  it('properly assigns `bg-danger` to `bgClass` if card loses', () => {
    const wrapper = shallow(
      <CardComponent
        winningCard="card1"
        card={{ image_uris: { current: 'bluewizard.jpg' } }}
        id="card2"
      />
    );
    const div = wrapper.find('div').first();
    expect(div.find('bg-danger')).toBeTruthy();
  });

  it('properly assigns `bg-secondary` to `bgClass` if no card wins', () => {
    const wrapper = shallow(
      <CardComponent
        winningCard=""
        card={{ image_uris: { current: 'bluewizard.jpg' } }}
        id="card2"
      />
    );
    const div = wrapper.find('div').first();
    expect(div.find('bg-secondary')).toBeTruthy();
  });
});
