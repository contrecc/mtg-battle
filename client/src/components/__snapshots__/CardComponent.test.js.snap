// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent renders correctly 1`] = `
<Card
  className="bg-success"
  style={
    Object {
      "marginBottom": "30px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": "0px",
      "maxWidth": "350px",
      "padding": "20px",
    }
  }
  tag="div"
>
  <GracefulImage
    alt="card image"
    className={null}
    height={null}
    noLazyLoad={false}
    noPlaceholder={false}
    noRetry={false}
    placeholderColor="#6c757d"
    retry={
      Object {
        "count": 10,
        "delay": 2,
      }
    }
    src="bluewizard.jpg"
    style={Object {}}
    width="100%"
  />
</Card>
`;
